src/error.test.js

```diff
@@ -1,10 +1,35 @@
-import { render, screen } from '@testing-library/react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { useGlobalContext } from './context';
+import { MemoryRouter } from 'react-router-dom';
 import Error from './Error';
+import Search from './Search';
 
 test('renders Error component', () => {
   render(<Error />);
   
   // Assert that the Error component is rendered
   const errorElement = screen.getByText(/Error/i);
   expect(errorElement).toBeInTheDocument();
-});
\ No newline at end of file
+});
+jest.mock('./context.js');
+test('renders search input and handles query change', () => {
+    const mockSetQuery = jest.fn();
+    const mockIsError = { show: false, msg: '' };
+    useGlobalContext.mockReturnValue({ query: '', setQuery: mockSetQuery, isError: mockIsError });
+
+    render(
+      <MemoryRouter>
+        <Search />
+      </MemoryRouter>
```

Comment:
Hi  @allahatim , 

Good job so far. !
There are some issues that you still need to work on to go to the next task  but you are almost there!
![mario_ladders](https://media.giphy.com/media/5hgYDDh5oqbmE4OKJ3/giphy.gif)



## Highlights
- Good testing approaches 
- Used unit testing and CI/CD pipeline 
- Clean testing code 

## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please make sure to separate the testing code of each functionality in a file
 - [ ] Please check the comments below 

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification